### PR TITLE
Change the icon by the achievement status

### PR DESCRIPTION
**Pull request by Laskow**
Change the icon displayed depending on the achievement status, locked or unlocked.

### CODE REVIEW

File: zscript/StupidAchievements.zs [hunk 1]
```diff
@@ -334,6 +341,8 @@ extend class sa_Achievement
   property margin        : margin;
   property border        : border;
   property isHidden      : isHidden;
+  property lockedIcon : lockedIcon;
+  property unlockedIcon : unlockedIcon;
```
Comment (mmaulwurff) [inline zscript/StupidAchievements.zs]: Align properties, please.

File: zscript/StupidAchievements.zs [hunk 2]
```diff
@@ -379,8 +390,11 @@ class sa_ // namespace
                       , int borderX, int borderY
                       , int borderWidth, int borderHeight
                       , int boxWidth, int boxHeight
+                      , int iconX, int iconY
+                      , int iconWidth, int iconHeight
                       , double alpha
                       , TextureID texture
+					   , TextureID Icon
```
Comment (mmaulwurff) [inline zscript/StupidAchievements.zs]: Here and in other places: please use spaces only, and preserve formatting.

File: zscript/StupidAchievements.zs [hunk 3]
```diff
@@ -473,6 +497,7 @@ class sa_Task abstract
   protected int       mNLines;
   protected Font      mFont;
   protected TextureID mTexture;
+  protected TextureID mIcon;
```
Comment (mmaulwurff) [inline zscript/StupidAchievements.zs:509]: mIcon is not initialized, no icon shows on on-screen achievement.

File: zscript/StupidAchievements.zs [hunk 4]
```diff
@@ -507,15 +532,23 @@ class sa_NoAnimationTask : sa_Task
     int borderX = x;
     int borderY = y;
 
+	int iconWidth = boxheight;
+	int iconHeight = iconWidth;
+	int iconX =  boxX - iconWidth * 1.1 ;
+	int iconY =  textY - mAchievement.margin / 2;
```
Comment (mmaulwurff) [inline zscript/StupidAchievements.zs]: Icon is put to the left of the text box, it won't be displayed if achievements are displayed on the left side of the screen.
In this case, icons should be either displayed to the right of the text box, or text box should be moved to the right.

**Review by mmaulwurff — CHANGES_REQUESTED** [review]
Unfortunately, I cannot accept graphics derived from copyrighted Doom graphics. I suggest making a placeholder icon.

We can include icons in a separate package.

File: zscript/StupidAchievements.zs (hunk 2 above)
Comment (Laskow) [reply zscript/StupidAchievements.zs]: These alignments are completely my mistakes. I fixed.

File: zscript/StupidAchievements.zs (hunk 4 above)
Comment (Laskow) [reply zscript/StupidAchievements.zs]: I know that the icon cannot be displayed in notification. Actually I wish to display it like Steam, GOG and other platforms, but I'm worried about it hides status bar and disturbs gameplay.

File: zscript/StupidAchievements.zs (hunk 3 above)
Comment (Laskow) [reply zscript/StupidAchievements.zs:509]: Can I ask how to fix it?

**Comment by Laskow** [comment]
> Unfortunately, I cannot accept graphics derived from copyrighted Doom graphics. I suggest making a placeholder icon.
> 
> We can include icons in a separate package.

I deleted all the icons that were in conflict with copyright. 
Also I added the new icon named "sa_icon.png" and changed all file names in "ImpAchievements.zs" (I should probably make an icon much better).

**Comment by Laskow** [comment]
Sorry, I'm a newbie to GitHub, I didn't know the button "re-request a review" on the right side...

By the way, should the icon be displayed in the notification? Let me know what you think.

File: zscript/StupidAchievements.zs (hunk 4 above)
Comment (mmaulwurff) [reply zscript/StupidAchievements.zs]: Even without the icon, notification obscures a part of the screen and disturbs gameplay. I think if a player doesn't like that, they can turn of notifications.

> By the way, should the icon be displayed in the notification? Let me know what you think.

So yes, I think it should.

File: zscript/StupidAchievements.zs (hunk 3 above)
Comment (mmaulwurff) [reply zscript/StupidAchievements.zs:509]: Assign a value in `sa_Task.init` like you do in `sa_AchievementItem.init`.

Comment (Laskow) [reply zscript/StupidAchievements.zs:509]: Thanks! I added some lines. Check the line between 386 – 393 and 497.

**Review by mmaulwurff — APPROVED** [review]

**Comment by mmaulwurff** [comment]
Thanks! I'll do some minor adjustments and release a new version of the library.

**Pull request merged**